### DISCUSSION TITLE
seems im not able to parse a json template with escaped quotes

### DISCUSSION THREAD

**Discussion by amassillo** [harttle/liquidjs#450] [Q&A]
Hi there!
Im looking for help :)
So I have a mongo document storing liquid markup expressions

`    "settings" : {
        "src" : "{{location.name}}",
        "href" : "visit us: {% if location.website !='' %}{{location.website}}{%else%}{{location.url}}{% endif %}",
        "exp" : "{% assign cf = location.tags | where: \"name\",\"Market\" | first %}{{cf.data}}"
    },`

When i stringify the whole node and try to parse it, im getting ""Error: tag "{% assign cf ..." not closed, probably related to the escape im doing for storing the expression on db
Any ideas about how could i get it working? If i process each key-val separately it works just fine

Thanks in advance
Ale

**Comment by harttle** [comment]
The top level syntax should be Liquid, that means the template passed into LiquidJS should be a valid Liquid template string. In this case the above string has a `if` tag and a `assign` tag, the `assign` tag is not valid:

```liquid
{% assign cf = location.tags | where: \"name\",\"Market\" | first %}
```

Note that the quotes are escaped (that's valid in JSON syntax, but not in Liquid syntax).

**Comment by awwit** [comment]
I solved this problem this way, but I'm sure there is a better solution:

```js
template = template.replace(/\{\{.*?\}\}/g, (match) => {
  return match.includes('\\"') ? JSON.parse(`"${match}"` : match
});
```
